app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersAction.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Hedonist\Actions\Place\GetPlaceCollectionByFilters;
+
+use Hedonist\Actions\Place\GetPlaceCollection\GetPlaceCollectionResponse;
+use Hedonist\Entities\Place\Location;
+use Hedonist\Repositories\Place\PlaceRepositoryInterface;
+use Hedonist\Repositories\Place\PlaceSearchCriteria;
+use Hedonist\Repositories\Review\Criterias\GetLastReviewByPlaceIdsCriteria;
+use Hedonist\Repositories\Review\ReviewRepositoryInterface;
+use Illuminate\Support\Facades\Auth;
+
+class GetPlaceCollectionByFiltersAction
+{
+    private $placeRepository;
+    private $reviewRepository;
+
+    public function __construct(
+        PlaceRepositoryInterface $placeRepository,
+        ReviewRepositoryInterface $reviewRepository)
+    {
+        $this->placeRepository = $placeRepository;
+        $this->reviewRepository = $reviewRepository;
+    }
+
+    public function execute(GetPlaceCollectionByFiltersRequest $request): GetPlaceCollectionResponse
+    {
+        $categoryId = $request->getCategoryId();
+        $location = $request->getLocation();
+        if (!empty($location) && preg_match('/.+,.+/', $location)) {
```

Comment:
if empty or incorrect format you can throw InvalidArgumentException. I would even parse it inside a request, and from the request return Location instance or throw InvalidArgumentException

Reply:
And change regexp like this /[0-9]+(\.[0-9]+)?,[0-9]+(\.[0-9]+)?/

Reply:
Done.
But I throw exceptions only when the wrong location format. With an empty location, I do not throw exceptions, because I can only search by category